src/contracts/evm/Contract.ts

```diff
@@ -0,0 +1,121 @@
+import { Interface } from "@ethersproject/abi";
+import {
+  SimulContractTx,
+  SimulSendNativeTx,
+  Tupleable,
+  TxLog,
+} from "types/evm";
+import { WalletState } from "contexts/WalletContext";
+import { toTuple } from "helpers";
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import Account from "./Account/Account";
+
+/** Represents generic contract functionality. Needs to be extended by actual contracts like {@link Account}. */
+export default class Contract {
```

Comment:
@misicnenad, @SovereignAndrey, 
If this structure is the right way to go ( which I believe is not with reasons mentioned in other comments ) we could just install `@ethersproject/contract` along with `@ethersproject/provider` or for this matter just install `ethers` as a whole (that is using this packages somehow doesn't manifest inefficiencies outlined in #1966)

There's no need for another PR parallel to #2025 just to re-create these functionalities.